### DISCUSSION TITLE
How to get 2d Gaussian Grid areas added to 2d diagnostic output files?

### DISCUSSION THREAD

**Discussion by aschuh** [NOAA-GFDL/GFDL_atmos_cubed_sphere#313] [Q&A] [answered]
It would be convenient to be able to generate the 2d grid areas of the Gaussian grid that UFS interpolates to, I'm having difficulty figuring out how to do that? Looking for a 192 x 384 array of surface areas basically, either in separate file or added to 2d sfc diagnostics. I tried adding:

grid spec
"dynamics", "area", "area", "grid_spec", "all", .false., "none", 2,

to diag_table but that doesn't seem to work. Ideas?

Second question might be how to extend back to native cubed sphere grid ... for "areas" for that? This all has to do w/ reconstructing mass budgets for atm tracers post hoc.

I also posted this over on the NOAA-UFS git repos: https://github.com/ufs-community/ufs-weather-model/discussions/2095

**Comment by lharris4** [comment, marked as answer]
Hi, Andrew. The areas on the remapped Gaussian grid would have to be written out from the regridding software or computed analytically. The area of a grid cell bounded by latitude and longitude circles is simple to compute: I believe it is $$R_e^2 \times \Delta \lambda \times \Delta \cos \phi.$$

On the native cubed-sphere grid the coordinates of the grid cell centroids, corners, and grid-cell areas can be written to history files by adding the following lines to your diag table (assuming the grid_spec file is not defined elsewhere in this file):
```
"grid_spec",              -1,  "months",   1, "days",  "time"
 "dynamics", "grid_lon", "grid_lon", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "grid_lat", "grid_lat", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "grid_lont", "grid_lont", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "grid_latt", "grid_latt", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "area",     "area",     "grid_spec", "all", .false.,  "none", 2,
```
See more about the diag_table format at [the FMS documentation](https://noaa-gfdl.github.io/FMS/group__diag__manager.html).

**Comment by bensonr** [comment]
@aschuh - What @lharris4 showed you will dump the native cubed-sphere information.  You can also get this same information from the input grid files.  The input grid files describe the supergrid, which is made up of the cell corners, cell-edge midpoints, and the cell centroid.  I will point out that the quantities (area, dx, dy, etc) in the input grid files are derived from the supergrid lat-lon pairs and are dependent on the definition of the R_e and pi.

I assume you are talking about the gaussian grid output from the UFS write component.  The definition of that grid occurs outside of FV3 and can be found in the UFS fv3atm [_write grid component_](https://github.com/NOAA-EMC/fv3atm/blob/develop/io/module_wrt_grid_comp.F90).  I don't see an area calculation, merely the definition of lat and lon.

**Comment by aschuh** [comment]
@bensonr- It is really cubed sphere "areas" I'm looking for in the output files.  It IS available internally.  I just want to be able to integrate up global atmos burden by multiplying mix ratio by pressure by area.  The cubed sphere isn't a regular grid so it's not clear how you'd use the topology, e.g. edges, corners, etc, to calc area.  I'd have to look at the Gaussian grid again, but I believe is still a GRID in lat/long just not regularly spaced.  Again, to do precise tracer mass checks, it would be nice if this is absolutely true.

From what I've heard, I think I'll progress like this. Try to figure out why adding :
"grid_spec",              -1,  "months",   1, "days",  "time"
 "dynamics", "grid_lon", "grid_lon", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "grid_lat", "grid_lat", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "grid_lont", "grid_lont", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "grid_latt", "grid_latt", "grid_spec", "all", .false.,  "none", 2,
 "dynamics", "area",     "area",     "grid_spec", "all", .false.,  "none", 2,
doesn't work for me, I'm assuming it should.  That would fix the "native grid" version of tracer mass summing.  For the Gaussian/interpolated, I'm guessing a grid based on grid_xt, grid_yt, calculated posthoc is what I'll use.

Anticipate questions coming soon on pressure levels : ). Thx all.

andrew
